network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/extensions/BranchObservabilityImpl.java

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.network.store.iidm.impl.extensions;
+
+import com.powsybl.commons.extensions.AbstractExtension;
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.extensions.BranchObservability;
+import com.powsybl.iidm.network.extensions.ObservabilityQuality;
+import com.powsybl.network.store.iidm.impl.AbstractBranchImpl;
+import com.powsybl.network.store.model.BranchObservabilityAttributes;
+import com.powsybl.network.store.model.ObservabilityQualityAttributes;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+public class BranchObservabilityImpl<B extends Branch<B>> extends AbstractExtension<B> implements BranchObservability<B> {
+
+    public BranchObservabilityImpl(B branch) {
+        super(branch);
+    }
+
+    private AbstractBranchImpl<?, ?> getBranch() {
+        return (AbstractBranchImpl<?, ?>) getExtendable();
+    }
+
+    private BranchObservabilityAttributes getBranchObservabilityAttributes() {
+        return (BranchObservabilityAttributes) getBranch().getResource().getAttributes().getExtensionAttributes().get(BranchObservability.NAME);
+    }
+
+    public boolean isObservable() {
+        return getBranchObservabilityAttributes().isObservable();
+    }
+
+    @Override
+    public BranchObservability<B> setObservable(boolean observable) {
+        getBranch().updateResource(res -> ((BranchObservabilityAttributes) res.getAttributes().getExtensionAttributes().get(BranchObservability.NAME)).setObservable(observable));
+        return this;
+    }
+
+    @Override
+    public ObservabilityQuality<B> getQualityP1() {
+        BranchObservabilityAttributes branchObservabilityAttributes = (BranchObservabilityAttributes) getBranch().getResource().getAttributes().getExtensionAttributes().get(BranchObservability.NAME);
```

Comment:
Is there a reason you don't use the method getBranchObservabilityAttributes everywhere in this class instead of redoing this ?

Reply:
Changed